src/components/box/box.pw.tsx

```diff
@@ -0,0 +1,897 @@
+
+  bgConstants.forEach(([color, rgbValue]) => {
+    test(`should verify background is ${color} when bg prop passed as RGB value`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<Default bg={rgbValue} />);
+      const boxElement = await getDataElementByValue(page, "box");
+      await expect(boxElement).toHaveCSS("background-color", rgbValue);
+    });
+  });
+
+  bgConstants.forEach(([color, rgbValue, hexValue]) => {
+    test(`should verify background is ${color} when bg prop passed as hex value`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<Default bg={hexValue} />);
+      const boxElement = await getDataElementByValue(page, "box");
+      await expect(boxElement).toHaveCSS("background-color", rgbValue);
+    });
+  });
+
+  bgConstants.forEach(([color, rgbValue]) => {
+    test(`should verify background is ${color} when backgroundColor prop passed as string`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<Default bg="" backgroundColor={color} />);
+      const boxElement = await getDataElementByValue(page, "box");
+      await expect(boxElement).toHaveCSS("background-color", rgbValue);
+    });
+  });
+
+  bgConstants.forEach(([color, rgbValue]) => {
+    test(`should verify background is ${color} when backgroundColor prop passed as RGB value`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<Default bg="" backgroundColor={rgbValue} />);
+      const boxElement = await getDataElementByValue(page, "box");
+      await expect(boxElement).toHaveCSS("background-color", rgbValue);
+    });
+  });
+
+  bgConstants.forEach(([color, rgbValue, hexValue]) => {
+    test(`should verify background is ${color} when backgroundColor prop passed as hex value`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<Default bg="" backgroundColor={hexValue} />);
+      const boxElement = await getDataElementByValue(page, "box");
+      await expect(boxElement).toHaveCSS("background-color", rgbValue);
+    });
+  });
+
+  ["0.1", "0.5", "1"].forEach((opacity) => {
+    test(`should verify opacity is ${opacity}`, async ({ mount, page }) => {
+      await mount(<Default bg="primary" opacity={opacity} />);
+      const boxElement = await getDataElementByValue(page, "box");
+      await expect(boxElement).toHaveCSS("opacity", opacity);
+    });
+  });
+
+  widthConstants.forEach(([width, percentage]) => {
+    test(`should verify width is ${width} when prop is passed as a percentage`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<Default bg="primary" width={percentage} />);
+      const boxElement = await getDataElementByValue(page, "box");
+      await expect(boxElement).toHaveAttribute("width", String(percentage));
+      await assertCssValueIsApproximately(boxElement, "width", parseInt(width));
+    });
+  });
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  widthConstants.forEach(([width, percentage, number]) => {
```

Comment:
**suggestion**: For parameterised tests where we have unused test parameters, what do you think about storing the parameters for each test in an object instead of an array? That way we can access the values we need without having to disable the linting rule for the unused ones.

```suggestion
  widthConstants.forEach(({ width, number }) => {
```  
where:
```ts
const widthConstants = [
  { width: "135px", percentage: 0.1, number: 135 },
  { width: "683px", percentage: 0.5, number: 683 },
  { width: "1366px", percentage: 1, number: 1366 },
];
```

Reply:
When I first read this comment I thought it was a good suggestion - but now I'm coming to actually address it I'm having significant doubts.

I certainly would rather not have to disable the linting, and destructuring objects rather than arrays looks quite nice too. But as you say, this would mean redefining `widthConstants` to be an array of objects rather than an array of arrays - which mean not just htis test, but the others that use `widthConstants`, would need changing from array to object destructuring. And this in turn would make the entire test suite quite inconsistent, in a way that could confuse future readers.

The entire file could be converted to use objects rather than arrays of course, but then that would be inconsistent with all the other conversions we've got going on (and completed), where we use arrays 😅 

For this reason I'm going to leave this as it is for now - but very happy to have a conversation about this.

Reply:
I've also just remembered that I can remove the ESLint warning without disabling it, if I change the destructuring to `[width, , number]`. This is valid syntax - although I have to admit, I find it pretty gross 😬 But I can live with it if you (and the other reviewer) prefer it!

Reply:
Cool thanks for this Robin 👍🏼 Personally I do prefer the object-based syntax as it is easier to read and handle typing, but happy to go with the majority if others disagree :)

Reply:
no, on reflection I think I do prefer your suggestion of using an array of object for at least some of these, and that the inconsistency isn't really a problem. But I'll have another think about this on Monday morning 😄